tun/netstack/tun.go

```diff
@@ -164,6 +164,10 @@ func (tun *netTun) Close() error {
 
 	tun.ep.Close()
 
+	if tun.incomingPacket != nil {
```

Comment:
What's the purpose of checking `!= nil` here? Does something else close it? If it is necessary, does that imply there's race potential?

Reply:
Gotcha, okay. Let me look closely.

Reply:
I guess the same applies to `tun.events`. So maybe I'll address both of those in a follow up.